lib/write_full.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-FileCopyrightText:  2023, Christian Göttsche <cgzones@googlemail.com>
+ *
+ * SPDX-License-Identifier:  BSD-3-Clause
+ */
+
+
+#include <config.h>
+
+#ident "$Id$"
+
+#include "prototypes.h"
+
+#include <errno.h>
+#include <unistd.h>
+
+
+/*
+ * write_full - write entire buffer
+ *
+ * Write up to count bytes from the buffer starting at buf to the
+ * file referred to by the file descriptor fd.
+ * Retry in case of a short write.
+ *
+ * Returns the number of bytes written on success, -1 on error.
+ */
+ssize_t write_full(int fd, const void *buf, size_t count) {
```

Comment:
Do we need to use `void *` in the second parameter?  I bet `const char *` would work.  That would add more type safety, and would avoid needing a cast in line 47.

Reply:
Well, I think he was just trying to keep the # of logical changes per commit to a minimum, which I appreciate.

I'll merge this, then a patch to make that change would be welcome.

Reply:
Yup, agree.  I hadn't realized that he didn't define this function from scratch, but rather moved it from somewhere else.  The commit message was rather misleading in that regard.